conjur/api/client.py

```diff
@@ -244,3 +252,33 @@ def change_personal_password(self, logged_in_user: str, current_password: str,
         """
         # pylint: disable=line-too-long
         return self._api.change_personal_password(logged_in_user, current_password, new_password)
+
+    def find_resources_by_identifier(self, resource_identifier: str) -> list:
+        """
+        Get all the resources with the given identifier.
+        """
+        list_constraints = {"search": resource_identifier}
+        returned_resources_ids = self.list(list_constraints)
+
+        def get_resource_kind_if_matches(returned_resource_id):
+            resource = Resource.from_full_id(returned_resource_id)
+            return resource if resource.identifier == resource_identifier else None
+
+        resources = map(get_resource_kind_if_matches, returned_resources_ids)
+        resources = [res for res in resources if res]  # Remove None elements
+        return resources
+
+    def find_resource_by_identifier(self, resource_identifier: str) -> list:
+        """
+        Look for a resource with the given identifier, and return its kind.
+        Fail if there isn't exactly one such resource.
+        """
+        resources = self.find_resources_by_identifier(resource_identifier)
+        if not resources:
+            raise ResourceNotFoundException(resource_identifier)
+        if len(resources) > 1:
+            raise MissingRequiredParameterException(
+                f"Ambiguous resource identifier, multiple resources exist with this identifier: "
+                f"({', '.join([res.full_id() for res in resources])})")
```

Comment:
@shulifink @szamir1 Please review text

Reply:
Not entirely sure what you ask, I will try to elaborate.
Resource identifier is the name in conjur, without the type (or 'kind'...).
That is, resource `layer:test/mylayer` is of type (kind) layer, and its identifier is `test/mylayer`.
There could be multiple resources with the same identifier, `layer:test`, `policy:test`, `variable:test`, so looking for the resource `test` is ambiguous, as it could be any of the previous resources.

Reply:
So can we say "Ambiguous resource identfier: `<identifier>`. There are multiple resrouces with this identifier."